test/BlockManager.js

```diff
@@ -900,5 +900,161 @@ describe('BlockManager', function () {
+        ['uint32', 'uint48[]', 'bytes32'],
+        [epoch, votes2, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+      );
+      await voteManager.connect(signers[13]).commit(epoch, commitment2);
+
+      const votes3 = [100, 200, 300, 400, 500, 600, 700, 800, 900];
+      const commitment3 = utils.solidityKeccak256(
+        ['uint32', 'uint48[]', 'bytes32'],
+        [epoch, votes3, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+      );
+      await voteManager.connect(signers[14]).commit(epoch, commitment3);
+
+      const votes4 = [100, 200, 300, 400, 500, 600, 700, 800, 900];
+      const commitment4 = utils.solidityKeccak256(
+        ['uint32', 'uint48[]', 'bytes32'],
+        [epoch, votes4, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+      );
+      await voteManager.connect(signers[15]).commit(epoch, commitment4);
+
+      const votes5 = [100, 200, 300, 400, 500, 600, 700, 800, 900];
+      const commitment5 = utils.solidityKeccak256(
+        ['uint32', 'uint48[]', 'bytes32'],
+        [epoch, votes5, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+      );
+      await voteManager.connect(signers[16]).commit(epoch, commitment5);
+
+      const votes6 = [100, 200, 300, 400, 500, 600, 700, 800, 900];
+      const commitment6 = utils.solidityKeccak256(
+        ['uint32', 'uint48[]', 'bytes32'],
+        [epoch, votes6, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+      );
+      await voteManager.connect(signers[17]).commit(epoch, commitment6);
+
+      await mineToNextState(); // reveal
+
+      await voteManager.connect(signers[12]).reveal(epoch, votes,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+
+      await voteManager.connect(signers[13]).reveal(epoch, votes2,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+
+      await voteManager.connect(signers[14]).reveal(epoch, votes3,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+
+      await voteManager.connect(signers[15]).reveal(epoch, votes4,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+
+      await voteManager.connect(signers[16]).reveal(epoch, votes5,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+
+      await voteManager.connect(signers[17]).reveal(epoch, votes6,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+
+      await mineToNextState(); // propose state
+
+      const stakerIdAcc12 = await stakeManager.stakerIds(signers[12].address);
+      let staker = await stakeManager.getStaker(stakerIdAcc12);
+      let { biggestInfluencerId } = await getBiggestInfluenceAndId(stakeManager);
+      let iteration = await getIteration(voteManager, stakeManager, staker);
+      await blockManager.connect(signers[12]).propose(epoch,
```

Comment:
Hey this looks good 
Just a suggestion 

this is dependent on value of maxAlt blocks 
right now its 5 so manually proposing 6 is okay 
if we change this then its issue

can we have 

base= 12
for loop till maxAlt blocks
  blockManager.connect(signers[base+i]).propose 
  
This will also reduce lot of redundant code.
We can also do same for commit and reveal.

Reply:
So here then loop variable (i) starts from 1 right ? @abhishekvispute  (12 already proposed, [12+1,12+2,12+3,12+4,12+5])

Reply:
yep

Reply:
ok will do that